scadalts-ui/src/views/DataObjects/Scripting/index.vue

```diff
@@ -1,33 +1,401 @@
+
+			this.selectedScript = null;
+			this.scriptForm.id = -1;
+			this.scriptForm.xid = '';
+			this.scriptForm.name = '';
+			this.scriptForm.pointsOnContext = [];
+			this.scriptForm.script = '';
+			this.scriptForm.datasourceContext = '';
+			this.scriptForm.datapointContext = 'dp';
+
+			this.dialog = true;
+		},
+		removeDatapoint(dataPointXid) {
+			this.scriptForm.pointsOnContext = this.scriptForm.pointsOnContext.filter(
+				(p) => p.dataPointXid != dataPointXid,
+			);
+		},
+
+		selectScript(id) {
+			this.selectedScriptId = id;
+			this.dialog = true;
+			this.selectedScript = this.scriptList.find((x) => x.id === this.selectedScriptId);
+			this.scriptForm.id = this.selectedScript.id;
+			this.scriptForm.xid = this.selectedScript.xid;
+			this.scriptForm.name = this.selectedScript.name;
+			this.scriptForm.pointsOnContext = this.selectedScript.pointsOnContext.map((x) => {
+				const dp = this.datapoints.find((dp) => dp.id === x.key);
+				return {
+					varName: x.value,
+					dataPointXid: dp.xid,
+				};
+			});
+
+			const oc = this.selectedScript.objectsOnContext;
+			if (oc && oc.length) {
+				const o1 = oc.find((x) => x.key == 1);
+				const o2 = oc.find((x) => x.key == 2);
+				if (o1 && o1.value) {
+					this.scriptForm.datasourceContext = o1.value;
+				} else {
+					this.scriptForm.datasourceContext = '';
+				}
+
+				if (o2 && o2.value) {
+					this.scriptForm.datapointContext = o2.value;
+				} else {
+					this.scriptForm.datapointContext = '';
+				}
+
+				this.scriptForm.datapointContext = o2.value;
+			} else {
+				this.scriptForm.datasourceContext = '';
+				this.scriptForm.datapointContext = '';
+			}
+
+			this.scriptForm.script = this.selectedScript.script;
+		},
+		addDatapoint() {
+			const p = this.datapoints.find((x) => x.id === this.selectedDatapointId);
+			if (!this.scriptForm.pointsOnContext.find((x) => p.xid === x.dataPointXid)) {
+				this.scriptForm.pointsOnContext.push({
+					dataPointXid: p.xid,
+					varName: `p${p.id}`,
+				});
+				this.selectedDatapointId = null;
+			}
+		},
+		async fetchScriptList() {
+			this.loading = true;
+			this.scriptList = await this.$store.dispatch('searchScripts', this.searchFilters);
+			if (!this.search) {
+				this.scriptListFiltered = this.scriptList;
+			} else {
+				const keywords = this.search.split(' ');
+				this.scriptListFiltered = this.scriptList.filter((x) =>
+					`${x.id} ${x.xid} ${x.name} ${x.script}`
+						.toLowerCase()
+						.includes(keywords[0].toLowerCase()),
+				);
+			}
+			this.totalScripts = this.scriptList.total;
+		},
+		runScript(xid) {
+			this.$store.dispatch('runScript', xid);
+			this.snackbar = true;
+			this.snackbarMessage = `${this.$t('scriptList.scriptExecuted')} `;
+		},
+		async saveScript() {
+			if (this.selectedScriptId != -1) {
+				response = await this.$store.dispatch('updateScript', this.scriptForm);
+			} else {
+				response = await this.$store.dispatch('createScript', this.scriptForm);
```

Comment:
After saving the script the "creation dialog" is still visible.

Reply:
fixed